### PR TITLE
fix kubectl set command

### PR DESCRIPTION
**Pull request by WanLinghao**
**What this PR does / why we need it**:

1. fix some errors in 'kubectl set' series command

2.remove unused function in kubectl set env command

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by WanLinghao** [comment]
@zjj2wry  PTAL

File: pkg/kubectl/cmd/set/set_env.go [hunk 1]
```diff
@@ -74,10 +74,10 @@ var (
 	  kubectl set env rc --all ENV=prod
 
 	  # Import environment from a secret
-	  kubectl set env --from=secret/mysecret dc/myapp
+	  kubectl set env --from=secret/mysecret deployment/myapp
```
Comment (zjj2wry) [inline pkg/kubectl/cmd/set/set_env.go]: why update this?

File: pkg/kubectl/cmd/set/set_env.go [hunk 2]
```diff
@@ -123,8 +123,7 @@ type EnvOptions struct {
 
 	Cmd *cobra.Command
 
-	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
```
Comment (zjj2wry) [inline pkg/kubectl/cmd/set/set_env.go]: should update `f.UpdatePodSpecForObject()`to `o.UpdatePodSpecForObject()` better

File: pkg/kubectl/cmd/set/set_env.go (hunk 1 above)
Comment (WanLinghao) [reply pkg/kubectl/cmd/set/set_env.go]: kubectl can not recognize dc as abbreviation of  deployment config

File: pkg/kubectl/cmd/set/set_env.go (hunk 2 above)
Comment (WanLinghao) [reply pkg/kubectl/cmd/set/set_env.go]: OK

File: pkg/kubectl/cmd/set/set_env.go (hunk 1 above)
Comment (WanLinghao) [reply pkg/kubectl/cmd/set/set_env.go]: @zjj2wry WDYT

Comment (zjj2wry) [reply pkg/kubectl/cmd/set/set_env.go]: yes, you are right

**Comment by zjj2wry** [comment]
@WanLinghao thank you do this.

**Comment by WanLinghao** [comment]
@fabianofranz @sttts PTAL

**Comment by WanLinghao** [comment]
@zjj2wry do you know who can help me push this patch, thanks a lot

**Comment by WanLinghao** [comment]
@BenTheElder do you know who can help me push this,  thanks a lot

**Comment by fabianofranz** [comment]
@WanLinghao thanks!
/approve no-issue
/sig cli

**Pull request merged**